packages/snaps-controllers/src/utils.ts

```diff
@@ -375,3 +376,68 @@ export function debouncePersistState(
     );
   };
 }
+
+/**
+ * Handlers allowed for tracking.
+ */
+export const TRACKABLE_HANDLERS = Object.freeze([
+  HandlerType.OnHomePage,
+  HandlerType.OnInstall,
+  HandlerType.OnNameLookup,
+  HandlerType.OnRpcRequest,
+  HandlerType.OnSignature,
+  HandlerType.OnTransaction,
+  HandlerType.OnUpdate,
+] as const);
+
+/**
+ * A union type representing all possible trackable handler types.
+ */
+export type TrackableHandler = (typeof TRACKABLE_HANDLERS)[number];
+
+/**
+ * Throttles event tracking calls per unique combination of parameters.
+ *
+ * @param fn - The tracking function to throttle.
+ * @param timeout - The timeout in milliseconds. Defaults to 60000 (1 minute).
+ * @returns The throttled function.
+ */
+export function throttleTracking(
+  fn: (
+    snapId: SnapId,
+    handler: TrackableHandler,
+    success: boolean,
+    origin: string,
+  ) => Promise<void>,
+  timeout = 60000,
+) {
+  const previousCalls = new Map<string, number>();
+
+  return async (
+    snapId: SnapId,
+    handler: TrackableHandler,
+    success: boolean,
+    origin: string,
+  ): Promise<void> => {
+    const key = `${snapId}${handler}${origin}`;
+    const now = Date.now();
+    const lastCall = previousCalls.get(key) ?? 0;
+
+    if (now - lastCall >= timeout) {
```

Comment:
Wondering if we can match existing implementation? Lodash `throttle` works a little differently:

```ts
import _ from 'lodash'
    
const f = _.throttle(console.log, 1000)

f(1); // Logs immediately
f(2); // Is ignored because of the next call
f(3); // Logs after 1000 ms
```